src/transformer/cli.js

```diff
@@ -53,28 +53,79 @@ export default async ({name, yargsOptions, callback}) => {
 		process.exit(-1);
 	}
 
-	const spinner = ora('Transforming records').start();
+	const spinner = ora(`Transforming${args.validate ? ' and validating' : ''}${args.fix ? ' and fixing' : ''} records`).start();
+
 	const options = {
 		stream: fs.createReadStream(args.file),
-		args,
-		spinner,
-		handleRecordsOutput
+		args
 	};
-	await callback(options);
 
-	function handleRecordsOutput(records) {
+	const TransformClient = callback(options);
+	let succesRecordArray = [];
+	let failedRecordsArray = [];
+	let counter = -1;
+	let numberOfRecords = 0;
+
+	await new Promise(resolve => {
+		TransformClient
+			.on('end', () => resolve(true))
```

Comment:
You can use a shorthand here: `.on('end', resolve)`

Reply:
if it does not have () => before resolve it runs it prematurely